OpenRA.Mods.TS/UtilityCommands/LegacyTilesetImporter.cs

```diff
@@ -57,7 +59,23 @@ public void Run(ModData modData, string[] args)
 				"Cliff" // TS defines this as "Rock"
 			};
 
+			var metadata = new StringBuilder();
+			metadata.AppendLine("General:");
+
+			var name = Path.GetFileNameWithoutExtension(args[1]);
+			if (name == "temperat")
+				name = "Temperate";
+
+			metadata.AppendLine("\tName: {0}".F(name));
+			metadata.AppendLine("\tId: {0}".F(name.ToUpperInvariant()));
+			metadata.AppendLine("\tPalette: iso{0}.pal".F(extension));
+			metadata.AppendLine("\tHeightDebugColors:  00000080, 00004480, 00008880, 0000CC80, 0000FF80, 4400CC80," +
+				" 88008880, CC004480, FF110080, FF550080, FF990080, FFDD0080, DDFF0080, 99FF0080, 55FF0080, 11FF0080");
+
 			// Loop over template sets
+			var data = new StringBuilder();
+			data.AppendLine("Templates:");
+			var categoriesUsage = new Dictionary<string, bool>();
```

Comment:
If you want to go this route (which is fine I guess) you should just use a `HashSet<string>`.